src/feed-item.tsx

```diff
@@ -1,58 +1,46 @@
 import React from 'react';
 import { Link } from 'react-router-dom';
-import { metadataService, ZnsMetadataService } from '@zer0-os/zos-zns';
+import { ZnsMetadataService } from '@zer0-os/zos-zns';
+import { augmentWithMetadata } from './util/feed';
+import { Model } from './feed-model';
 import CloudImage from './components/cloud-image';
 
 import './styles.css';
 
-export interface Model {
-  id: string;
-  title: string;
-  description: string;
-  imageUrl: string;
-  znsRoute: string;
-  metadataUrl?: string;
-}
-
-export interface Properties extends Model {
+export interface Properties {
+  item: Model;
   app: string;
-  metadataService?: ZnsMetadataService;
+  metadataService: ZnsMetadataService;
+  metadataAbortController: AbortController;
+  setSelectedItem: (item: Model) => void;
 }
 
-interface State extends Model {}
+interface State {
+  item: Model;
+}
 
 export class FeedItem extends React.Component<Properties, State> {
-  constructor(props) {
-    super(props);
+  state = { item: this.props.item };
 
-    const { app, metadataService, ...model } = props;
+  componentDidMount = async () => {
+    const { item, metadataService, metadataAbortController } = this.props;
 
-    this.state = {
-      ...model,
-    };
+    this.setState({ item: await augmentWithMetadata(item, metadataService, metadataAbortController) });
```

Comment:
there are a few different things on this line that point to something deeper. the root of it, though, is that we should not be loading data inside a component. the solution here, is that this component should just take props, and render them.